native/src/main/java/io/ballerina/lib/sap/Client.java

```diff
@@ -183,13 +212,52 @@ private static void setImportParams(JCoParameterList jcoPramList, BMap<BString,
+                            entry.getValue())).getElementType()).getTag() == TypeTags.BYTE_TAG) {
+                        jcoPramList.setValue(entry.getKey().toString(),
+                                ValueCreator.createArrayValue((byte[]) entry.getValue()));
+                    }
+                    break;
+                case TypeTags.RECORD_TYPE_TAG:
+                    if ("Date".equals(TypeUtils.getType(entry.getValue()).getName())) {
+                        BMap<BString, Object> dateMap = (BMap<BString, Object>) entry.getValue();
+
+                        Object yearObj = dateMap.get(StringUtils.fromString("year"));
+                        Object monthObj = dateMap.get(StringUtils.fromString("month"));
+                        Object dayObj = dateMap.get(StringUtils.fromString("day"));
+                        Object hourObj = dateMap.get(StringUtils.fromString("hour"));
+                        Object minuteObj = dateMap.get(StringUtils.fromString("minute"));
+                        Object secondObj = dateMap.get(StringUtils.fromString("second"));
+
+                        if (yearObj != null && monthObj != null && dayObj != null) {
+                            int year = Integer.parseInt(yearObj.toString());
+                            int month = Integer.parseInt(monthObj.toString());
+                            int day = Integer.parseInt(dayObj.toString());
+
+                            int hour = (hourObj != null) ? Integer.parseInt(hourObj.toString()) : 0;
+                            int minute = (minuteObj != null) ? Integer.parseInt(minuteObj.toString()) : 0;
+                            int second = (secondObj != null) ? Integer.parseInt(secondObj.toString()) : 0;
+
+                            Calendar calendar = Calendar.getInstance();
+                            calendar.set(year, month - 1, day, hour, minute, second);
+                            Date dateValue = calendar.getTime();
+                            jcoPramList.setValue(entry.getKey().toString(), dateValue);
+                        } else {
+                            throw SAPErrorCreator.fromBError("Invalid date record: year, month, and day " +
+                                    "must be provided.", null);
+                        }
+                    }
+                    break;
                 default:
                     throw SAPErrorCreator.fromBError("Error while setting input parameter for field: " +
                             entry.getKey().toString() + ". Unsupported type " + type + " Supported types are: " +
                             "string, int, float, decimal", null);
             }
         });
     }
-
+    private static void throwUnsupportedUnionTypeError(Object key) {
+        throw SAPErrorCreator.fromBError("Error while setting input parameter for field: " +
+                key.toString() + ". Unsupported union type. Supported types are: string, int, float, decimal", null);
+    }
```

Comment:
```suggestion
        throw SAPErrorCreator.fromBError("Unsupported union type '" + + "' for field '" +
                key.toString() + '". Supported types are: string, int, float, decimal and nil able supported types. ", null);
    }
```

Reply:
Add the coming type in the error message as well.

Reply:
Changed with [010b6b2](https://github.com/ballerina-platform/module-ballerinax-sap.jco/pull/12/commits/010b6b2c237f70b1e7db75e29c8c780db22968de).